cedar-policy-core/src/est/err.rs

```diff
@@ -91,6 +93,27 @@ pub enum FromJsonError {
     /// Error reported when the extension function name is unknown
     #[error("Invalid extension function name: `{0}`")]
     UnknownExtFunc(ast::Name),
+    /// Returned when an Entity UID used as an action does not have the type `Action`
+    #[error(transparent)]
+    #[diagnostic(transparent)]
+    InvalidActionType(#[from] InvalidActionType),
+}
+
+/// Details about an `InvalidActionType` error.
+#[derive(Debug, Diagnostic, Error)]
+#[diagnostic(help("action entities must have type `Action`, optionally in a namespace"))]
+pub struct InvalidActionType {
+    pub(crate) euids: NonEmpty<Arc<crate::ast::EntityUID>>,
+}
+
+impl std::fmt::Display for InvalidActionType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "expected that action entity uids would have the type `Action` but got "
+        )?;
+        join_with_conjunction(f, "and", self.euids.iter(), |f, e| write!(f, "`{e}`"))
```

Comment:
Minor: maybe we want to print entity types instead of EUIDs? Maybe say something like `Entities A::a, B::b, C::c should have type `Action` but got `A`, `B`, `C`.